ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryBatchedLedgerReaderWriter.scala

```diff
@@ -126,24 +118,56 @@ object InMemoryBatchedLedgerReaderWriter {
           metrics,
           inStaticTimeMode = needStaticTimeModeFor(timeProvider))
       val validator = BatchedSubmissionValidator[Index](
-        BatchedSubmissionValidatorParameters.default,
+        BatchedSubmissionValidatorFactory.defaultParametersFor(
+          batchingLedgerWriterConfig.enableBatching),
         keyValueCommitting,
         new ConflictDetection(metrics),
         metrics,
-        engine)
-      Resource.successful(
+        engine
+      )
+      val committer =
+        BatchedValidatingCommitter[Index](
+          () => timeProvider.getCurrentTime,
+          validator,
+          stateValueCache)
+      val readerWriter =
```

Comment:
Am I right to assume that the InMemoryBatchedLedgerReaderWriter is in fact a generic Reader/Writer because it still needs to be wrapped in BatchedSubmissionValidator here:
```
BatchedSubmissionValidatorFactory.batchingLedgerWriterFrom(
            batchingLedgerWriterConfig,
            readerWriter)
```
In that case, would it not be better to drop `Batched` from its name?

Reply:
Yes, as I wrote it in the description I'll do that in a separate PR (this one only deletes the old classes).